helper/resource/error.go

```diff
@@ -0,0 +1,21 @@
+package resource
+
+type NotFoundError struct {
+	LastError    error
+	LastRequest  interface{}
+	LastResponse interface{}
+	Message      string
+	Retries      int
+}
+
+func (e *NotFoundError) Error() string {
+	if e.Message != "" {
+		return e.Message
+	}
+
+	return "couldn't find resource"
+}
+
+func NewNotFoundError(err string) *NotFoundError {
```

Comment:
For consistency with `resource.RetryableError(err)` & friends, perhaps change this to:
```go
func NotFoundError(err error) *NotFoundError {
  return &NotFoundError{LasError: err}
}
```
For ease of adding a message, you could also add:
```go
func (e *NotFoundError) WithMessage(msg string) *NotFoundError {
  e.Message = msg
  return e
}
```
Invocation would look like:
```go
return resource.NotFoundError(err).WithMessage("Lambda Permissions couldn't be found")
```